src/main/java/com/codeus/winter/util/PropertySource.java

```diff
@@ -0,0 +1,82 @@
+package com.codeus.winter.util;
+
+public abstract class PropertySource<T> {
+    private final String name;
+
+    private final T source;
+
+    protected PropertySource(String name, T source) {
+        hasText(name);
```

Comment:
There are methods for the same purpose in the Objects.requireNotNull() and String.isBlank().

Reply:
I did add to method 'hasText' checking for String.isBlank()